### PR TITLE
feat: update crowdin workflow and add pretranslate step

### PR DESCRIPTION
**Pull request by nikaaru**
# Summary

crowdin was removed from the publish and check workflows.
From now on, we only have crowdin in the release process
So that for each release:

- First, it extracts the new strings
- Then it pushes new strings to crowdin
- Then the pre-translate script will be called
- Then it downloads the translations for different languages, and opens a PR and merges all new translations to the **Next** branch.
And finally, it continues the release process

# How did you test this change?

To test, you can first take some PRs from **Next** Branch and change the string in some files and merge them into **Next** Branch.
Finally, call the release workflow manually,
In this case, the new strings that are on the next branch must be uploaded to crowdin, pre-translated and new translations must be downloaded for each language and merged in the **Next** branch. And the release process continues

# Checklist:

- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Implemented a user interface (UI) change, referencing our Figma design to ensure pixel-perfect precision.

### CODE REVIEW

File: scripts/crowdin/pretranslate.mjs [hunk 1]
```diff
@@ -0,0 +1,159 @@
+const projectId = process.env.CROWDIN_PROJECT_ID;
```
Comment (yeager-eren) [inline scripts/crowdin/pretranslate.mjs:1]: This script should be structured like other scripts which means the entry point for running the command will be `command.mjs` (you can keep your helpers here).

Comment (yeager-eren) [reply scripts/crowdin/pretranslate.mjs:1]: In first lines of your main function in `command.mjs`, please make sure your checking envs and fails early if they are not exist.

File: scripts/crowdin/pretranslate.mjs [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+const token =  process.env.CROWDIN_PERSONAL_TOKEN;
+
+const BASE_URL = `https://api.crowdin.com/api/v2/projects/${projectId}`;
+const preTranslateURL = `${BASE_URL}/pre-translations`;
+const REQUEST_INTERVAL_TIMEOUT = 10_000;
+const MAXIMUM_REQUEST_RETRIES = 12;
```
Comment (yeager-eren) [inline scripts/crowdin/pretranslate.mjs:7]: Naming is a bit misleading. I suggest something like `MAXIMUM_PRETRANSLATATION_STATUS_CHECK` or `MAXIMUM_STATUS_CHECK` or `MAX_STATUS_CHECK`

File: scripts/crowdin/pretranslate.mjs [hunk 3]
```diff
@@ -0,0 +1,159 @@
+const projectId = process.env.CROWDIN_PROJECT_ID;
+const token =  process.env.CROWDIN_PERSONAL_TOKEN;
+
+const BASE_URL = `https://api.crowdin.com/api/v2/projects/${projectId}`;
+const preTranslateURL = `${BASE_URL}/pre-translations`;
```
Comment (yeager-eren) [inline scripts/crowdin/pretranslate.mjs:5]: Uppercase please. it's constant, not a normal variable.

Comment (yeager-eren) [reply scripts/crowdin/pretranslate.mjs:5]: same comment applies to `token` and `projectId` as well.

File: scripts/crowdin/pretranslate.mjs [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+const REQUEST_INTERVAL_TIMEOUT = 10_000;
+const MAXIMUM_REQUEST_RETRIES = 12;
+
+const requestData = {
+  method: 'mt',
+  engineId: 410374,
```
Comment (yeager-eren) [inline scripts/crowdin/pretranslate.mjs:11]: Don't use hardcoded value, engineId seems different in each account. you wrote `getMachineTranslationEngineID` but you are not using it

File: scripts/crowdin/pretranslate.mjs [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+  autoApproveOption: 'all',
+  duplicateTranslations: false,
+  skipApprovedTranslations: true,
+  translateUntranslatedOnly: true,
+  translateWithPerfectMatchOnly: false,
+};
+
+const getMachineTranslationEngineID = async () =>{
+  try {
+    const response = await fetch('https://api.crowdin.com/api/v2/mts', {
```
Comment (yeager-eren) [inline scripts/crowdin/pretranslate.mjs:21]: Why some of api endpoinst are defined as const but here you've used hardcoded value directly?

File: scripts/crowdin/pretranslate.mjs [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+      throw new Error(`Failed to get source file id`);
+    }
+
+    const responseData = await response.json();
+    const sourceFileId = responseData.data[0].data.id;
+
+    return sourceFileId;
+  } catch (error) {
+    console.error('Error:', error);
+    throw error;
```
Comment (yeager-eren) [inline scripts/crowdin/pretranslate.mjs:82]: Use same approach as we did in `scripts/common/errors.mjs` for errors. You can add CrowdinError for instance and use it for your errors here.

Comment (yeager-eren) [reply scripts/crowdin/pretranslate.mjs:82]: Same comment applies to other errors.

**Review by yeager-eren — CHANGES_REQUESTED** [review]
Thanks. I did my review. I have one question as well. Why we didn't use official crowdin's npm packages? I didn't dig into it but seems there are some packages out there.

**Comment by yeager-eren** [comment]
I also getting bad request on my testing, Do I'm missing something here?
https://github.com/yeager-eren/rango-client/actions/runs/7938518813

File: scripts/crowdin/pretranslate.mjs (hunk 1 above)
Comment (nikaaru) [reply scripts/crowdin/pretranslate.mjs]: Fixed

File: scripts/crowdin/pretranslate.mjs (hunk 2 above)
Comment (nikaaru) [reply scripts/crowdin/pretranslate.mjs]: Fixed

File: scripts/crowdin/pretranslate.mjs (hunk 3 above)
Comment (nikaaru) [reply scripts/crowdin/pretranslate.mjs]: Fixed

File: scripts/crowdin/pretranslate.mjs (hunk 4 above)
Comment (nikaaru) [reply scripts/crowdin/pretranslate.mjs]: Fixed

File: scripts/crowdin/pretranslate.mjs (hunk 5 above)
Comment (nikaaru) [reply scripts/crowdin/pretranslate.mjs]: Fixed

File: scripts/crowdin/pretranslate.mjs (hunk 6 above)
Comment (nikaaru) [reply scripts/crowdin/pretranslate.mjs]: FIxed

**Pull request merged**